pkg/sql/logictest/testdata/logic_test/alter_table

```diff
@@ -663,3 +663,23 @@ query I
 SELECT COUNT(*) FROM crdb_internal.jobs WHERE status = 'pending' OR status = 'started'
 ----
 0
+
+# Can add and drop column in same transaction
+statement ok
+BEGIN; CREATE TABLE add_drop_column (id INT); ALTER TABLE add_drop_column ADD COLUMN column_to_drop INT; ALTER TABLE add_drop_column DROP COLUMN column_to_drop; COMMIT;
```

Comment:
I'd use table "t" for both these tests

Reply:
Also add a schema change that adds more than one column

ALTER TABLE add_drop_column ADD COLUMN column_that_stays INT, column_to_drop INT

Reply:
You also want to test this with a UNIQUE column and ensure that the implicit index is dropped.